client/session.go

```diff
@@ -3753,3 +3860,30 @@ func (v FetchBlocksMetadataEndpointVersion) String() string {
 	}
 	return "unknown"
 }
+
+func newTagsFromEncodedTags(
+	encodedTags checked.Bytes,
+	tagDecoder serialize.TagDecoder,
+	idPool ident.Pool,
+) (ident.Tags, error) {
+	if encodedTags == nil {
+		return nil, nil
+	}
+
+	encodedTags.IncRef()
+	tagDecoder.Reset(encodedTags)
+
+	tags := make(ident.Tags, 0, tagDecoder.Remaining())
```

Comment:
Could you re-use the `storage/index.TagArrayPool` here? would have to pipe from sessionPools all the way to the `block.Metadata.Finalize()` method I think

Reply:
I think we'd have to create a wrapper type and plumb it through here.  Let's fix this async:
https://github.com/m3db/m3db/issues/614